File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int n = nums.length;
        int []result= new int[n];

        int l = 0, r = n - 1, i = n - 1;
        while (l <= r) {
            int l_sq = nums[l] * nums[l];
            int r_sq = nums[r] * nums[r];
            if (l_sq > r_sq) {
                result[i--] = l_sq;
                l++;
            } else {
                result[i--] = r_sq;
                r--;
            }
        }
        
        return result;  
    }
}

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        l,r = 0, len(nums)-1
        result = [0]* len(nums)
        for i in range (len(nums)-1,-1,-1):
            if (abs(nums[l])> abs(nums[r])):
                result[i] = nums[l]* nums[l]
                l+=1
            else:
                result[i] = nums[r]*nums[r]
                r-=1
        return result